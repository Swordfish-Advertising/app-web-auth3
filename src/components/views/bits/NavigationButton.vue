<template>
    <v-btn
      id="navButton"
      @click="goTo(page)"
    >{{title}}</v-btn>
</template>
<script>
  export default {
    props: ['title', 'page'],
    methods: {
      goTo (page) {
        this.$router.push(page)
      }
    }
  }
</script>