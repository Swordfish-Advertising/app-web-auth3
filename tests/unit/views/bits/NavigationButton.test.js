/* global describe, it, expect */

import { shallowMount, mount, createLocalVue } from '@vue/test-utils';
import NavigationButton from '@/components/views/bits/NavigationButton';
import Vue from 'vue';
import Vuetify from 'vuetify';
import VueRouter from 'vue-router';

Vue.use(Vuetify);

describe('NavigationButton.test.js', () => {
  it('renders correctly (snapshots matching)', () => {
    const wrapper = shallowMount(NavigationButton);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('has the correct title and goes to the correct page', () => {
    const title = 'buttonTitle';
    const page = 'pageId';
    const newPath = '/' + page;
    const localVue = createLocalVue();
    localVue.use(VueRouter);
    const router = new VueRouter();
    const wrapper = mount(NavigationButton, {
      propsData: {
        title: title,
        page: page,
      },
      localVue,
      router,
    });
    const button = wrapper.find('#navButton');
    expect(button.text()).toBe(title);
    expect(wrapper.vm.$route.path).toBe('/');
    button.trigger('click');
    expect(wrapper.vm.$route.path).toBe(newPath);
  });
});
