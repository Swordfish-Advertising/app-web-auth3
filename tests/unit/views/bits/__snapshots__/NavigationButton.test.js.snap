// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationButton.test.js renders correctly (snapshots matching) 1`] = `
<vbtn-stub
  activeclass="v-btn--active"
  id="navButton"
  ripple="true"
  tag="button"
  type="button"
>
  
</vbtn-stub>
`;
